packages/jest-transform/src/ScriptTransformer.ts

```diff
@@ -475,7 +476,7 @@ export default class ScriptTransformer {
         }
       },
       {
-        exts: [path.extname(moduleName)],
+        exts: ext === '' ? ['.js'] : [ext],
```

Comment:
@SimenB @jeysal Prevent the hook from getting applied when `moduleName` refers to a directory.
Any comments on this? I'll separate this if the approach is ok

Reply:
I think we'd need moduleFileExtensions? Not sure if I have a full overview of the effect of this, I'm on mobile only for the next 10 days. @simenb can you take a look?

Reply:
Mapping over `moduleFileExtensions` makes sense (to add the leading dot)

Reply:
A separate pr, yes please 🙂

Reply:
`moduleFileExtensions` looks like the best option, I'll run test the tests over all the transform related PRs to see if there is any impact there